### ISSUE TITLE
Problems reported pasting from Word

### ISSUE THREAD

**Issue by catflinger** [fifteensquared/fifteensquared-theme#13]
Email by Geoff reporting this:

I have just copied/pasted from Word into the WP Editor and the font size was far larger than that which it should have been, except for the text to which I had added a link which was correctly sized. When previewing the post, the font size was also far larger than normal apart from the linked text, very undesirable!
 
Copying/pasting from Notepad produced the correct font size. In both cases the text style was Paragraph.

**Comment by catflinger** [comment]
I have tried cut-and-paste form Word 2016 on both sandbox and live sites and the editor and preview look normal. 

Please can you be specific about what element of the post is not sowing properly?  Eg is it a table, heading, fixed width font etc...

**Comment by Walahfrid** [comment]
In case it helps, I copied/pasted from Word into the sandbox editor on Saturday (2/12) and all was fine. I have just done so again with no problems. I see that you were trying the same thing at the same time. The sandbox seems fine, the live site isn’t.

**Comment by Walahfrid** [comment]
I tried the same blog that was OK in the sandbox on the live site and it was OK there as well. I tried a repeat of my much shorter earlier announcement and had the same result as before. All the body text was larger than it should have been except for the text containing the link. A repeat without the link gave the same result, all the body text was enlarged.  The Word font size in all cases was the same as I always use, 12pt Arial.

**Comment by catflinger** [comment]
Can you get this down to a minimal Word document that shows the error?  Perhaps start a document containing just one short sentence in 12pt and cut and paste that.  If that works keep adding one line with just one new feature (eg a link) until it breaks.

**Comment by Walahfrid** [comment]
I have now repeated the tests in the sandbox. A long blog has the correct sized font in the WP editor, A short 2-line post has the font size increased. So, the live site and the sandbox are the same.

**Comment by Walahfrid** [comment]
@catflinger 
That is what I have been doing for the last 15 minutes. No luck so far.  The single line you suggest as a starting point has an oversized font in the editor after paste, as does 2 to 6 lines, I have yet to find out if/when the font size returns to as it should be. I cannot see why the font size should be post length related but will continue testing it.

**Comment by catflinger** [comment]
The cause of this is that Word is adding in-line styles to the text that it pastes.  Because they are in-line these styles override the styling of the site.  The site css has the font at 0.9em but word overrides it to be medium.

The markup pasted by Word is 
`<span style="color: #000000; font-size: medium;">Here are two more lines.</span>
`
This will always appear black and medium sized no matter what the CSS is in the site or the editor.

I don't think this is a bug with the styling in the editor as it also appears too large in the published blog.

Is this related to the Wordpress version rather than the wp-andreas225 theme?

**Comment by Walahfrid** [comment]
It seems to be too inconsistent for that. My findings attached.
[Editor font tests.pdf](https://github.com/fifteensquared/fifteensquared-theme/files/1531301/Editor.font.tests.pdf)

**Comment by catflinger** [comment]
I don't think this has anything to do with the styling in the editor.  

I think this is an issue that has to be taken up with Wordpress support.

**Comment by catflinger** [comment]
I have removed the styling from the editor in the sandbox.  Please try again one more time.  If the problem persists then this is an issue for the Wordpress support team to investigate.

**Comment by Walahfrid** [comment]
Results as before on retest. Having seen the changes you made to editor-style.css I'm very surprised that I was still getting the modified fonts in the editor as opposed to the default ones. I will raise this issue with WordPress support but doubt that it will be resolved in the near future. Therefore I suggest removing editor-style.css and returning to the default WP editor. I will try this first in the sandbox.

**Comment by catflinger** [comment]
I had to do a hard reset on my browser cache to get the old styling back.  

I'm sure whether we have styling in the editor view on or off is irrelevant.  This is related to Word 2003.  When I paste form Word 2016 I don't get any of the in-line styles generated that you are seeing in your posts from Word 2003.

**Comment by Walahfrid** [comment]
I cleared my browsers cache before the retest. I had no problems whatsoever with Word 2003/WP editor prior to the addition of editor-style.css so I still suggest that we revert to the default WP editor by removing this file.

**Comment by catflinger** [comment]
It would be interesting to install a clipboard viewer, such as http://www.freeclipboardviewer.com and see what is getting pasted from your Word 2003 document.  We could compare that to what gets pasted form my Word 2016.

**Comment by catflinger** [comment]
The styling has already been removed from the sandbox.  Editor-styling.css is still there but it is now empty.

If you are still seeing the styling in the editor then your cache clearing hasn't worked or else something else is doing the caching.

I used Chrome, enabled developer tools and then did a right-click on the refresh button and selected Empty Cache and Hard reload.  I got the old editor back immediately.

**Comment by Walahfrid** [comment]
I will try that shortly (I'm also trying to monitor the site while all this is going on). In the meantime I have attached the results from the clipboard viewer (in HTML format).
Wasn't allowed to upload the file (not supported) so have changed its extension from .cl7 to .txt
[Word Clipboard Output HTML.txt](https://github.com/fifteensquared/fifteensquared-theme/files/1531837/Word.Clipboard.Output.HTML.txt)

**Comment by Walahfrid** [comment]
Font now back to default in the sandbox editor (perhaps I should have closed down then reopened IE11 after clearing the history, the only significant change since my last visit). Repeated the previous test (one line, two line etc) and no change to the editor font at all when pasting from Word 2003. The font size change issue has definitely been caused by the addition of editor-style.css.

I am thinking that when set to default (ie no editor-style.css) the editor strips out any Word formatting, apart from the likes of bold and italics, but when 'styled' it deliberately allows more formatting to get through (though this doesn't explain why the font size is increased with some pastes but not with others).

**Comment by catflinger** [comment]
I can't find anything wrong here.  

The clipboard html generated by Word 2003 contains a span that has font-size 12.0pt using the Micrsoft Office specific styling elements (prefix mso-...)
```
<span style='font-size:12.0pt;font-family:Arial;mso-fareast-font-family:"Times New Roman";
mso-ansi-language:EN-GB;mso-fareast-language:EN-GB;mso-bidi-language:AR-SA'>1
Idyll chap's ruined? <b style='mso-bidi-font-weight:normal'><i
style='mso-bidi-font-style:normal'>Easy thing to do</i></b> (6,4) </span>
```
In the same post I added a span myself at 12.0pt using the standard styling elements

`<span style="font-size: 12.0pt;">This is 12 pt</span>`

Both of these look the same size as each other and both larger then the font in the rest of the blog.  Wordpress seems to be faithfully reproducing exactly what MS Word instructed it to.

**Comment by catflinger** [comment]
Lets do a clean and simple test.  The live site has editor styling, the sandbox does not.  Using the same browser that has had its cache cleared open two windows, one in the live site one in the sandbox.

In each window begin a new post.  Cut and paste the same problematic text from Word into each window identically.  Don't do or add anything else to the post.

Save each post as a draft and then preview them both.

Do the two post previews look identical?  If not how do they differ?

**Comment by Walahfrid** [comment]
Duly done. The text sample I used was:

Everyman 3,711
One thing I like about Everyman puzzles is that there are never any cryptic definitions and very few double definitions. This week was no exception.

When pasted into the sandbox the editor font and post font were normal.
When pasted into the live site the editor and post fonts were normal if I only copied up to the full stop.
When pasted into the live site the editor and post fonts were enlarged if I copied up to the end of the second line (ie the 'return' was included). Copying in Word always includes the 'return' if one quickly selects the text to be copied by dragging the mouse pointer over the required lines.

**Comment by catflinger** [comment]
Too much information!  Lets keep it really really really simple.  Use the smallest one-line document.

What happens if we just do a single identical copy once into each window?  No extra text, no extra copies.  Don't even go back to Word, reuse the same clipboard content in each window.  Do the absolute minimum identically in each window.  Ignore what it looks like in the editor.  

Are the previews the same?

**Comment by Walahfrid** [comment]
No. Live site enlarged, sandbox normal.

**Comment by catflinger** [comment]
Good.  That is a great result.

Next test, I reinstate the editor styling on the sandbox.  We then repeat the same test identically.   Clear browser caches, use the same Word document as last time (if you still have it) copy the same text, same procedures, no extra tests, and no deviations.

I will replay when editor styling is back.

**Comment by catflinger** [comment]
OK, sandbox has editor styling again.

Repeat the same simple test exactly.  

If both previews are the same then we have nailed the problem.

**Comment by Walahfrid** [comment]
Font now enlarged in both the live site and the sandbox.

**Comment by catflinger** [comment]
Thanks for all that.  There is definitely something amiss and we should remove the styling from the editor.  

In the short term I will create a patch for editor-config.css that removes all the content.  It should be safe to deploy this single file to live with minimal re-testing.

The next build that gets a proper release test can have the feature either completely missing or completely fixed as the case may be.

**Comment by Walahfrid** [comment]
Paul
It shouldn't be necessary to create a patch, I can simply delete editor-style.css on the live site. This will not affect the rest of the theme and is what I suggested in my original email (we could have saved a lot of time today which could have been spent on more productive things).

One of my concerns was that some other bloggers, with less experience of WordPress, may encounter a font size problem with the method they use to create a post and not know how to rectify/prevent it.

I can remember back to when Tilsit (Dave Tilley) was still a blogger. I don't know how he added his last few posts but the font size and line spacing was all over the place. Not just a simply enlarged font but many font sizes, some that must have been in excess of 30pt, seemingly randomly occurring.

**Comment by catflinger** [comment]
Don't just delete the file from live!  Please.

functions.php attempts to use this file and if it is not there that might cause an error.  I will create a css file that has no effect.  This way the file is still a valid stylesheet and can be read and loaded.  It just doesn't do anything.

This is much the safest way and requires the minimum of retesting.  

In a future release we can change functions.php so the file is not loaded any more.  Then you can delete it.

**Comment by Walahfrid** [comment]
OK. I was not aware that the new functions.php references editor-style.css. I understand why you tried to add this enhancement, but was it worth the hassle?

**Comment by catflinger** [comment]
The sandbox now contains a reduced editor-style.css that does not do anything.  If you are happy this works OK in the sandbox you can copy it across to the live system.

**Comment by Walahfrid** [comment]
Sandbox editor has reverted to default font. No error or debug log entries when adding a new post. I will copy editor-style.css to the live site when all of today's blogs have been posted.

**Comment by Walahfrid** [comment]
The new editor-style.css has been copied to the live site.

**Comment by catflinger** [comment]
Patched in live site v1.0.1 + patch01

removed entirely from v1.0.2 onwards
